### ISSUE TITLE
phpactor status run fails

### ISSUE THREAD

**Issue by SelenZzz** [phpactor/phpactor#2490]
I've just installed phpactor using PHAR Installation. Calling "phpactor status" fails with following text:

> In Process.php line 355:
> 
> Unable to launch a new process.

<img width="439" alt="Screenshot 2024-01-06 at 12 24 30" src="https://github.com/phpactor/phpactor/assets/58228495/0c624253-01f5-4bf8-b422-6d22ae47691a">

**Comment by dantleech** [comment]
does anything else work? can you provide the stack trace with `phpactor status -vvv`?

**Comment by SelenZzz** [comment]
I'm running phpactor on MacOS with fish and bash terminal.

I've tried `class:reflect`, but got the following:
```
server:master* λ phpactor class:reflect ./Common/Arrays.php

In ClassNameCandidates.php line 39:

  There are no class name candidates


class:reflect [--format FORMAT] [--] <name>
```

`phpactor status -vvv` gives next:
```
server:master* λ phpactor status -vvv

Box Requirements Checker
========================

> Using PHP 8.3.0
> PHP is using the following php.ini file:
  /opt/homebrew/etc/php/8.3/php.ini

> Checking Box requirements:
  ✔ The application requires a version matching "^8.1".
  ✔ The application requires the extension "zlib".
  ✔ The application requires the extension "pcntl".
  ✔ The application requires the extension "posix".
  ✔ The package "amphp/dns" requires the extension "filter".
  ✔ The package "amphp/dns" requires the extension "json".
  ✔ The package "ramsey/uuid" requires the extension "json".
  ✔ The package "amphp/socket" requires the extension "openssl".
  ✔ The package "kelunik/certificate" requires the extension "openssl".
  ✔ The package "symfony/service-contracts" conflicts with the extension "psr".


 [OK] Your system is ready to run the application.


[FPR    ][DEBU][541283.642163] Resolved path "%project_root%" to "/Users/yermac/CloudStation/GitHub/test-v2/server"
[SCF    ][WARN][541283.642794] Filesystem "git" not supported: "The cwd does not seem to be a git repository root (could not find .git folder)"
[FPR    ][DEBU][541283.642808] Resolved path "%project_root%" to "/Users/yermac/CloudStation/GitHub/test-v2/server"
[FPR    ][DEBU][541283.642820] Resolved path "%project_root%" to "/Users/yermac/CloudStation/GitHub/test-v2/server"
[FPR    ][DEBU][541283.642834] Resolved path "%project_root%/vendor/autoload.php" to "/Users/yermac/CloudStation/GitHub/test-v2/server/vendor/autoload.php"
[FPR    ][DEBU][541283.643732] Resolved path "%project_root%" to "/Users/yermac/CloudStation/GitHub/test-v2/server"
[FPR    ][DEBU][541283.643755] Resolved path "%project_root%/composer.json" to "/Users/yermac/CloudStation/GitHub/test-v2/server/composer.json"

In Process.php line 355:

  [Symfony\Component\Process\Exception\RuntimeException]
  Unable to launch a new process.


Exception trace:
  at phar:///Users/yermac/.local/bin/phpactor/vendor/symfony/process/Process.php:355
 Symfony\Component\Process\Process->start() at phar:///Users/yermac/.local/bin/phpactor/vendor/symfony/process/Process.php:252
 Symfony\Component\Process\Process->run() at phar:///Users/yermac/.local/bin/phpactor/lib/Extension/Core/Application/Status.php:73
 Phpactor\Extension\Core\Application\Status->check() at phar:///Users/yermac/.local/bin/phpactor/lib/Extension/Core/Command/StatusCommand.php:24
 Phpactor\Extension\Core\Command\StatusCommand->execute() at phar:///Users/yermac/.local/bin/phpactor/vendor/symfony/console/Command/Command.php:298
 Symfony\Component\Console\Command\Command->run() at phar:///Users/yermac/.local/bin/phpactor/vendor/symfony/console/Application.php:1040
 Symfony\Component\Console\Application->doRunCommand() at phar:///Users/yermac/.local/bin/phpactor/vendor/symfony/console/Application.php:301
 Symfony\Component\Console\Application->doRun() at phar:///Users/yermac/.local/bin/phpactor/lib/Application.php:48
 Phpactor\Application->doRun() at phar:///Users/yermac/.local/bin/phpactor/vendor/symfony/console/Application.php:171
 Symfony\Component\Console\Application->run() at phar:///Users/yermac/.local/bin/phpactor/bin/phpactor:45
 require() at /Users/yermac/.local/bin/phpactor:15

status
```

**Comment by dantleech** [comment]
are you running it in your project root, is the project root a git repository?

**Comment by SelenZzz** [comment]
I figured it out.

My project uses composer, but autoload wasn't configured in my `composer.json`.
Also git repository isn't located in project root (but as turned out, changing this isn't required for fix).

Adding the lines below and running `composer dump-autoload` fixes the issue. 
```
  "autoload": {
    "psr-4": {
      "": "./"
    }
  }
``` 

Perhaps phpactor should handle the case when these lines are missing in `composer.json`.

Thanks for participation

**Comment by dantleech** [comment]
thanks -- I can run `phpactor status` in a folder without git (e.g. `/`). Is the error due to the missing composer lines or git?

**Comment by SelenZzz** [comment]
`phpactor class:reflect` runs fine after adding missing `composer.json` lines. 

`phpactor status` still fails to run.

**Comment by dantleech** [comment]
> phpactor status still fails to run.

even in a folder with git?

the code that's failing is in Symfony and tbh I don't know what could cause it to happen: 

```
        $this->process = @proc_open($commandline, $descriptors, $this->processPipes->pipes, $this->cwd, $envPairs, $this->options);

        if (!\is_resource($this->process)) {
            throw new RuntimeException('Unable to launch a new process.');
        }
```

**Comment by SelenZzz** [comment]
Tested it with completely new repository. 
Running commands below causes the same error:
```
GitHub        λ mkdir test-v3
GitHub        λ cd test-v3
test-v3       λ git init
test-v3:main  λ composer init -n --name "yermac/test-v3" --autoload "src"
test-v3:main* λ phpactor status

In Process.php line 355:

  Unable to launch a new process.
```


Also I've tried check if proc_open works fine on my system. 
Ran a modified example script from https://www.php.net/manual/en/function.proc-open.php
```
$descriptorspec = array(
    0 => array("pipe", "r"),
    1 => array("pipe", "w"),
    2 => array("file", "/tmp/error-output.txt", "a"),
);

$cwd = '/tmp';
$env = array('some_option' => 'aeiou');

$process = proc_open('ls ~/CloudStation/GitHub/test-v3/', $descriptorspec, $pipes, $cwd, $env);

if (is_resource($process)) {
    fwrite($pipes[0], '<?php print_r($_ENV); ?>');
    fclose($pipes[0]);

    echo stream_get_contents($pipes[1]);
    fclose($pipes[1]);

    $return_value = proc_close($process);

    echo "command returned $return_value\n";
}
```
```
test-v3:main* λ php src/test.php
composer.json
src
vendor
command returned 0
test-v3:main* λ
```
Output looks fine.

**Comment by SelenZzz** [comment]
> The __DIR__ line will not work with the PHAR

Manual installation fixes the issue. Thank you

**Comment by dantleech** [comment]
this should have been fixed in #2536
